lib/libutee/arch/arm/ftrace/ftrace.c

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+#include <string.h>
+#include <user_ta_header.h>
+#include <utee_syscalls.h>
+
+#define FTRACE_RETFUNC_DEPTH		50
+
+static unsigned long ftrace_ret_stack[FTRACE_RETFUNC_DEPTH];
+static unsigned int ftrace_ret_idx;
+static const char hex_str[] = "0123456789abcdef";
+
+void __ftrace_return(void);
+void ftrace_enter(unsigned long pc, unsigned long *lr);
+
+static void __noprof ftrace_memmove(struct ftrace_buf *fbuf, size_t size)
+{
+	char *dst = (char *)fbuf + fbuf->buf_off;
+	const char *src = (char *)fbuf + fbuf->buf_off + size;
+	size_t len = 0;
+
+	fbuf->curr_size -= size;
+	len = fbuf->curr_size;
+
+	while (len--)
+		*dst++ = *src++;
+}
+
+static size_t __noprof to_func_enter_fmt(char *buf, unsigned long pc)
+{
+	char *str = buf;
+	uint32_t i, addr_size = 2 * sizeof(unsigned long);
```

Comment:
prefer 1 line per local variable.
And always default initialize each local variable.

Reply:
ok. BTW, isn't it generic practice to initialize index variable as part of loop?

Reply:
Yes, but a looks a good practice to always init local variables where define, so that we don't bother when toolchain reporting un-initialized variable even when not applicable. Therefore, here the recommendation is:
```c
	int index = 0;

	for (i = 0; i < something ; i++)
		(...)
```
Compiler optims should make `index` be initialized only once.

Reply:
ok